examples/example_uq_bootstrap.py

```diff
@@ -0,0 +1,131 @@
+
+In this example, we demonstrate how to perform uncertainty quantification (UQ) using
+bootstrap method. We use a Stillinger-Weber (SW) potential for silicon that is archived
+in OpenKIM_.
+
+For simplicity, we only set the energy-scaling parameters, i.e., ``A`` and ``lambda`` as
+the tunable parameters. These parameters will be calibrated to energies and forces of a
+small dataset, consisting of 4 compressed and stretched configurations of diamond silicon
+structure.
+"""
+
+
+##########################################################################################
+# To start, let's first install the SW model::
+#
+#    $ kim-api-collections-management install user SW_StillingerWeber_1985_Si__MO_405512056662_006
+#
+# .. seealso::
+#    This installs the model and its driver into the ``User Collection``. See
+#    :ref:`install_model` for more information about installing KIM models.
+
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from kliff.calculators import Calculator
+from kliff.dataset import Dataset
+from kliff.loss import Loss
+from kliff.models import KIMModel
+from kliff.models.parameter_transform import LogParameterTransform
+from kliff.uq.bootstrap import BootstrapEmpiricalModel
+from kliff.utils import download_dataset
+
+##########################################################################################
+# Before running bootstrap, we need to define a loss function and train the model. More
+# detail information about this step can be found in :ref:`tut_kim_sw` and
+# :ref:`tut_params_transform`.
+
+# Instantiate a transformation class to do the log parameter transform
+param_names = ["A", "lambda"]
+
+# Create the model
+model = KIMModel(model_name="SW_StillingerWeber_1985_Si__MO_405512056662_006")
+
+# Set the tunable parameters and the initial guess
+opt_params = {"A": [["default"]], "lambda": [["default"]]}
+
+model.set_opt_params(**opt_params)
+model.echo_opt_params()
+
+# Get the dataset
+dataset_path = download_dataset(dataset_name="Si_training_set_4_configs")
+# Read the dataset
+tset = Dataset(dataset_path)
+configs = tset.get_configs()
+
+# Create calculator
+calc = Calculator(model)
+# Only use the forces data
+ca = calc.create(configs, use_energy=False, use_forces=True)
+
+# Instantiate the loss function
+residual_data = {"normalize_by_natoms": False}
+loss = Loss(calc, residual_data=residual_data)
+
+# Train the model
+min_kwargs = dict(method="lm")  # Optimizer setting
+loss.minimize(**min_kwargs)
+model.echo_opt_params()
+
+##########################################################################################
+# To perform UQ by bootstrapping, the general workflow starts by instantiating
+# :class:`~kliff.uq.bootstrap.BootstrapEmpiricalModel`, or
+# :class:`~kliff.uq.bootstrap.BootstrapNeuralNetworkModel` if using a neural network
+# potential.
+
+
+# It is a good practice to specify the random seed to use in the calculation to generate
+# a reproducible simulation.
+np.random.seed(1717)
```

Comment:
I don't love this way to specific seed. Can we implement it as `BootstrapEmpiricalModel(..., seed=<default>)` and provide a default?

Reply:
Is the concern here that the class can affect the numpy-wide generator? In that case, we can create a local random state and provide and make this an optional argument that user can specify. Any thoughts?

Reply:
I was thinking a bit simpler than that -- just moving `np.random.seed(<seed>)` into the `BootstripEmpiricalModel` class. 

It is more cosmetic than functional.